browser/conversion-tests/conversion_test.ts

```diff
@@ -0,0 +1,1003 @@
+}
+
+// The interval (in milliseconds) to check whether Wasm is done loading.
+const wasmLoadCheckIntervalMs = 100;
+
+// We use a global variable to track whether the Wasm code has finished loading.
+let isWasmLoaded = false;
+const loadEventName = '0xmeshtest';
+window.addEventListener(loadEventName, () => {
+    isWasmLoaded = true;
+});
+
+// Start compiling the WebAssembly as soon as the script is loaded. This lets
+// us initialize as quickly as possible.
+const go = new Go();
+WebAssembly.instantiate(wasmBuffer, go.importObject)
+    .then(module => {
+        go.run(module.instance);
+    })
+    .catch(err => {
+        // tslint:disable-next-line no-console
+        console.error('Could not load Wasm');
+        // tslint:disable-next-line no-console
+        console.error(err);
+        // If the Wasm bytecode didn't compile, Mesh won't work. We have no
+        // choice but to throw an error.
+        setImmediate(() => {
+            throw err;
+        });
+    });
+
+/*********************** Tests ***********************/
+// tslint:disable:custom-no-magic-numbers
+// tslint:disable:no-console
+
+(async () => {
+    await waitForLoadAsync();
+    const contractEvents = await conversionTestCases.contractEventsAsync();
+    testContractEvents(contractEvents);
+    const getOrdersResponse = await conversionTestCases.getOrdersResponseAsync();
+    testGetOrdersResponse(getOrdersResponse);
+    const orderEvents = await conversionTestCases.orderEventsAsync();
+    testOrderEvents(orderEvents);
+    const signedOrders = await conversionTestCases.signedOrdersAsync();
+    testSignedOrders(signedOrders);
+    const stats = await conversionTestCases.statsAsync();
+    testStats(stats);
+    const validationResults = await conversionTestCases.validationResultsAsync();
+    testValidationResults(validationResults);
+
+    // This special #jsFinished div is used to signal the headless Chrome driver
+    // that the JavaScript code is done running.
+    const finishedDiv = document.createElement('div');
+    finishedDiv.setAttribute('id', 'jsFinished');
+    document.querySelector('body')!.appendChild(finishedDiv); // tslint:disable-line:no-non-null-assertion
+})().catch(err => {
+    throw err;
+});
+
+function prettyPrintTestCase(name: string, description: string): (section: string, value: boolean) => void {
+    return (section: string, value: boolean) => {
+        console.log(`(${name} | ${description} | ${section}): ${value}`);
+    };
+}
+
+function testContractEvents(contractEvents: WrapperContractEvent[]): void {
+    // ERC20ApprovalEvent
+    let printer = prettyPrintTestCase('contractEventTest', 'ERC20ApprovalEvent');
+    testContractEventPrelude(printer, contractEvents[0]);
+    printer('kind', contractEvents[0].kind === 'ERC20ApprovalEvent');
+    const erc20ApprovalParams = contractEvents[0].parameters as WrapperERC20ApprovalEvent;
+    printer('parameter | owner', erc20ApprovalParams.owner === hexUtils.leftPad('0x4', 20));
```

Comment:
optional idea: How about `parameters.owner` to more clearly show which part of the object we are testing for equality? As it is currently written, we would have to do a sort of mental translation between this format and how the TypeScript type is actually structured. This idea applies in a few other places as well: anywhere there is a direct mapping to a property of a TypeScript object.

```suggestion
    printer('parameters.owner', erc20ApprovalParams.owner === hexUtils.leftPad('0x4', 20));
```

Reply:
I like that a lot more!